internal/p2p/transport_mconn.go

```diff
@@ -138,19 +138,35 @@ func (m *MConnTransport) Accept(ctx context.Context) (Connection, error) {
 		return nil, errors.New("transport is not listening")
 	}
 
-	tcpConn, err := m.listener.Accept()
-	if err != nil {
+	conCh := make(chan net.Conn)
```

Comment:
If we add 1 buffer slot for each of these channels, we can simplify some of the context checking below (since the writes will always succeed, and thus we needn't select)

Reply:
I feel like this approach, while I now understand it's reasonable, always feels... harder to read? 

Regardless, for some reason this has broken something terrifying in unrelated code, so I'm going to defer thinking about it.

Reply:
That's reasonable. I generally find it easy to understand one-buffered channels for one-to-one coordination. It's a fairly common idiom, though I'd stop short of saying "standard".  That said, I'm not too exercised about it. I find it easier to reason about fewer channel ops per select, on balance, but either works.